blackbird_python/blackbird/tests/test_auxiliary.py

```diff
@@ -197,6 +205,96 @@ def test_function_cos(self, parser, ctx, n, expected, num):
         expression = num(n)
         assert _func(func, expression) == np.cos(expected)
 
+    @pytest.mark.parametrize('n, expected', test_floats)
+    def test_function_tan(self, parser, ctx, n, expected, num):
+        """Test that a Blackbird tan function is properly called"""
+        func = blackbirdParser.FunctionContext(parser, ctx)
+        func.TAN = lambda: True
+        expression = num(n, num_type='float')
+        assert _func(func, expression) == np.tan(expected)
+
+    def test_function_arcsin(self, parser, ctx, num):
+        """Test that a Blackbird arcsin function is properly called"""
+        n = "0.543"

```

Comment:
How come some of these are tested for multiple values and others for a single value?

Reply:
Some of the functions are very picky with allowed values, and I suddenly thought that using huge amounts of test data isn't necessary - a single value passing should indicate it is the correct function.

Reply:
I think I will standardize this in a new PR.

Reply:
:+1: